### ISSUE TITLE
x86/64 lifting loop instruction

### ISSUE THREAD

**Issue by sangkilc** [B2R2-org/B2R2#18]
**Describe the bug**
The `loop` instruction contains unnecessary internal (intra) jumps as discussed in #15.

**To Reproduce**
```
0x19146D:    E2 FB                          loop -0x3 ; 0x19146A
```

By lifting the above instruction, we get:

```
-------------ISMark (19146D, 2)-------------
T_710:I32 := ECX
-------------LMark (Loop)-------------
T_710:I32 := (T_710:I32 - 0x1:I32)
if(T_710:I32 != 0x0:I32) then Jmp (Continue, 127) else Jmp (End, 128)
-------------LMark (Continue)-------------
EIP := (EIP + 0x19146A:I32)
Jmp (Loop, 126)
-------------LMark (End)-------------
-------------IEMark (19146F)-------------
```

**Expected behavior**
The `loop` instruction in x86 should be handled the same as in `jcc` instructions. But currently we are treating `loop` in a particularly different way.

**Additional context**

See #15 for more discussion.

**Comment by sijung07** [comment]
Fixed in be647ce1d.

**Comment by sangkilc** [comment]
Confirmed.

**Comment by enkomio** [comment]
Hi,

now the loop instruction is correctly lifted as:

    0x19146D:    E2 FB                          loop -0x3 ; 0x19146A


    type: ISMark     => -------------ISMark (19146D, 2)-------------
    type: Put        => T_710:I32 := (ECX - 0x1:I32)
    type: InterCJmp  => EIP := if (T_710:I32 != 0x0:I32) then 0x19146A:I32 else 0x19146F:I32
    type: IEMark     => -------------IEMark (19146F)-------------

Thanks!

**Comment by sangkilc** [comment]
Oh wow.  You are right.

**Comment by sangkilc** [comment]
@enkomio Would you be able to create a PR for this?

**Comment by enkomio** [comment]
I'll try for sure :)

**Comment by sangkilc** [comment]
Merged #19
